Automattic/wp-calypso
# Media: Fix dropdown filter not working on some screen widths

Fix #23878

The [`SectionNavTabs`](https://github.com/Automattic/wp-calypso/blob/17276793273dd19f31a6e52de675fb8222918ff4/client/components/section-nav/tabs.jsx) component has three way to render its children:
- As actual tabs, when the container is wide enough to display all the labels.
- As a [`SelectDropdown`](https://github.com/Automattic/wp-calypso/tree/17276793273dd19f31a6e52de675fb8222918ff4/client/components/select-dropdown) when the container is not wide enough.
- As a special kind of dropdown for mobile screens.

`SectionNavTabs` expects its children to be [`SectionNavTabItem`](https://github.com/Automattic/wp-calypso/blob/17276793273dd19f31a6e52de675fb8222918ff4/client/components/section-nav/item.jsx) with (in this case) the `onClick` prop.

Currently though, the Media `SectionNavTabs` uses [`FilterItem`](https://github.com/Automattic/wp-calypso/blob/17276793273dd19f31a6e52de675fb8222918ff4/client/my-sites/media-library/filter-item.jsx) as children with the sole purpose of calling the click handler with `props.value` as parameter.
For some reason, the `FilterItem`'s click handler is called `onChange` which, though, is not the `onClick` that `SelectDropdown` expects.

This PR removes `FilterItem` altogether and replaces it with a simple `SectionNavTabItem`.
The `changeFilter` function becomes a higher-order function that can be called right away instead of requiring the whole "middleware" component.

## Testing instructions

- Open `/media` on a screen large enough that the navigation bar displays all the items as tabs.
- Make sure that clicking on those items changes the content of the page.
- Reduce the screen width so that the items are grouped in a dropdown button.
- Again test clicking the navigation items.
- Reduce the screen width as much as possible in order to show a mobile layout.
- Test the navigation items one last time.
